pkg/util/crypto.go

```diff
@@ -96,31 +94,53 @@ func (kms SecretsKMS) DeletePassphrase(key string) error {
 	return nil
 }
 
+// GetID is returning empty string
+func (kms SecretsKMS) GetID() string {
```

Comment:
For secrets based KMS, if we return an empty string the following will occur,
- We will store an empty string in the OMap key for KMS [here](https://github.com/ceph/ceph-csi/pull/822/files?file-filters%5B%5D=.go#diff-a6cef7e66fbdf22f7065bf167606d504R244-R245)
- When using VolID based context building, we would retrieve it as an empty string, and not set encrypted property for the volune [here](https://github.com/ceph/ceph-csi/pull/822/files?file-filters%5B%5D=.go#diff-8e27516879efe971bf4755fadf585e06R355-R362)

From an testing standpoint, expanding this volume would have succeeded even though this is encrypted (also, creating a snapshot from this encrypted image will succeed). This is because these are the places where the `genVolFromVolID` is used.

We could leverage presence of the key, but absence of value (or empty value), to denote secrets based encryption.
OR,
As before, we choose a reserved name for the `kmsID` when it is defaulting to secrets, and note it in the doc. as a reserved name. (maybe, "csi-secrets", as this is from the CSI secrets field that we get the required data).

Reserving a name is always tricky, but I am borderline fine with it if others have no objections.

Reply:
good point actually, I'll update to actually store the value for default kms also